src/geometry/manifold/algebra/left_invariant_derivation.lean

```diff
@@ -0,0 +1,176 @@
+  derivation.eval_at g to_derivation f)
+
+variables {I G}
+
+namespace left_invariant_derivation
+
+instance : has_coe (left_invariant_derivation I G) (derivation 𝕜 C^∞⟮I, G; 𝕜⟯ C^∞⟮I, G; 𝕜⟯)
+:= ⟨λ X, X.to_derivation⟩
+
+instance : has_coe_to_fun (left_invariant_derivation I G) := ⟨_, λ X, X.to_derivation.to_fun⟩
+
+variables
+{M : Type*} [topological_space M] [charted_space H M] {x : M}
+{X Y : left_invariant_derivation I G} {f f' : C^∞⟮I, G; 𝕜⟯} {r : 𝕜}
+
+lemma to_fun_eq_coe : X.to_fun = ⇑X := rfl
+lemma coe_fn_coe : ⇑(X : C^∞⟮I, G; 𝕜⟯ →ₗ[𝕜] C^∞⟮I, G; 𝕜⟯) = X := rfl
+@[simp] lemma to_derivation_eq_coe : X.to_derivation = X := rfl
+
+lemma coe_injective (h : ⇑X = Y) : X = Y :=
+by { cases X, cases Y, congr', exact derivation.coe_injective h }
+
+@[ext] theorem ext (h : ∀ f, X f = Y f) : X = Y :=
+coe_injective $ funext h
+
+variables (X Y f)
+
+@[simp] lemma coe_lift_eq_coe :
+  ⇑(X : derivation 𝕜 C^∞⟮I, G; 𝕜⟯ C^∞⟮I, G; 𝕜⟯) = (X : C^∞⟮I, G; 𝕜⟯ → C^∞⟮I, G; 𝕜⟯) := rfl
+
+/-- Premature version of the lemma. Prefer using `left_invariant` instead. -/
+lemma left_invariant' : (𝒅(𝑳 I g)) (1 : G) (derivation.eval_at (1 : G) ↑X) f =
+  derivation.eval_at g ↑X f := by rw [←to_derivation_eq_coe]; exact left_invariant'' X f g
+
+instance : has_zero (left_invariant_derivation I G) := ⟨⟨0, λ f g,
+  by simp only [linear_map.map_zero, derivation.coe_zero]⟩⟩
+
+instance : inhabited (left_invariant_derivation I G) := ⟨0⟩
+
+instance : has_add (left_invariant_derivation I G) :=
+{ add := λ X Y, ⟨X + Y, λ f g, by simp only [linear_map.map_add, derivation.coe_add,
+  left_invariant', pi.add_apply]⟩ }
+
+instance : has_neg (left_invariant_derivation I G) :=
+{ neg := λ X, ⟨-X, λ f g, by simp only [linear_map.map_neg, derivation.coe_neg, left_invariant',
+  pi.neg_apply]⟩ }
+
+@[simp] lemma coe_add : ⇑(X + Y) = X + Y := rfl
+@[simp] lemma coe_zero : ⇑(0 : left_invariant_derivation I G) = 0 := rfl
+@[simp] lemma coe_neg : ⇑(-X) = -X := rfl
+@[simp, norm_cast] lemma lift_add :
+  (↑(X + Y) : derivation 𝕜 C^∞⟮I, G; 𝕜⟯ C^∞⟮I, G; 𝕜⟯) = X + Y := rfl
+@[simp, norm_cast] lemma lift_zero :
+  (↑(0 : left_invariant_derivation I G) : derivation 𝕜 C^∞⟮I, G; 𝕜⟯ C^∞⟮I, G; 𝕜⟯) = 0 := rfl
+@[simp] lemma map_add : X (f + f') = X f + X f' := is_add_hom.map_add X f f'
+@[simp] lemma map_zero : X 0 = 0 := is_add_monoid_hom.map_zero X
+@[simp] lemma map_neg : X (-f) = -X f := linear_map.map_neg X f
+@[simp] lemma map_sub : X (f - f') = X f - X f' := linear_map.map_sub X f f'
+
+instance : add_comm_group (left_invariant_derivation I G) :=
+{ add_assoc := λ X Y Z, ext $ λ a, add_assoc _ _ _,
+  zero_add := λ X, ext $ λ a, zero_add _,
+  add_zero := λ X, ext $ λ a, add_zero _,
+  add_comm := λ X Y, ext $ λ a, add_comm _ _,
+  add_left_neg := λ X, ext $ λ a, add_left_neg _,
+  ..left_invariant_derivation.has_zero,
+  ..left_invariant_derivation.has_add,
+  ..left_invariant_derivation.has_neg }
+
+@[simp] lemma coe_sub : ⇑(X - Y) = X - Y := by simp only [sub_eq_add_neg, coe_add, coe_neg]
```

Comment:
This feels like it might be convenient to have true by `rfl` - do you think it's worth defining `sub` manually such that this is true?

Reply:
I have not idea but I will do it if you think it is a good idea!